apps/ckit-app/src/hooks/useUnipass.tsx

```diff
@@ -0,0 +1,22 @@
+import { UnipassWallet } from '@ckit/ckit';
+import { useLocalStorage } from '@rehooks/local-storage';
+
+interface UnipassContext {
+  shouldLogin: boolean;
+  cacheLogin: () => void;
+  shouldSendTx: boolean;
+  cacheTx: (tx: string | null) => void;
+  cachedTx: string | null;
+}
+
+export function useUnipass(): UnipassContext {
+  const [cachedTx, cacheTx] = useLocalStorage<string | null>('unipassTx');
+  const adapter = new UnipassWallet.UnipassRedirectAdapter({ host: 'https://unipass.xyz' }); // TODO  use the config
+  return {
+    shouldLogin: adapter.hasLoginInfo(),
+    cacheLogin: () => adapter.saveLoginInfo(),
+    shouldSendTx: adapter.hasSigData(),
+    cacheTx,
+    cachedTx,
```

Comment:
Shouldn't we automatically clear the cache after the transaction is sent?

Reply:
cleared